src/apis/searchTextFull.js

```diff
@@ -22,8 +22,93 @@ WebViewer(...)
  */
 
 import actions from 'actions';
+import core from "core";
+import { getSearchListeners } from 'helpers/search';
 
-export default store => (searchValue, options) => {
-  store.dispatch(actions.openElement('searchPanel'));
-  store.dispatch(actions.searchTextFull(searchValue, options));
-};
+function buildSearchModeFlag(options = {}) {
+  const SearchMode = core.getSearchMode();
+  let searchMode = SearchMode.PAGE_STOP | SearchMode.HIGHLIGHT;
+
+  if (options.caseSensitive) {
+    searchMode |= SearchMode.CASE_SENSITIVE;
+  }
+  if (options.wholeWord) {
+    searchMode |= SearchMode.WHOLE_WORLD;
+  }
+  if (options.wildcard) {
+    searchMode |= SearchMode.WILD_CARD;
+  }
+  if (options.regex) {
+    searchMode |= SearchMode.REGEX;
+  }
+
+  searchMode |= SearchMode.AMBIENT_STRING;
+
+  return searchMode;
+}
+
+export default function searchTextFull(dispatch) {
+  return function searchTextFull(searchValue, options) {
+    if (dispatch) {
+      // dispatch is only set when doing search through API (instance.searchText())
+      // When triggering search through UI, then redux updates are already handled inside component
+      dispatch(actions.openElement('searchPanel'));
+      dispatch(actions.searchTextFull(searchValue, options));
+    }
+
+    const searchMode = buildSearchModeFlag(options);
+
+    let hasActiveResultBeenSet = false;
+    function onResult(result) {
+      core.displayAdditionalSearchResult(result);
+      if (!hasActiveResultBeenSet) {
+        // when full search is done, we make first found result to be the active result
+        core.setActiveSearchResult(result);
+        hasActiveResultBeenSet = true;
+      }
+    }
+
+    function searchInProgressCallback(isSearching) {
+      // execute search listeners when search is complete, thus hooking functionality search in progress event.
+      if (isSearching === false) {
+        const results = core.getPageSearchResults();
+        const searchOptions = {
+          // default values
+          caseSensitive: false,
+          wholeWord: false,
+          wildcard: false,
+          regex: false,
+          searchUp: false,
+          ambientString: true,
+          // override values with those user gave
+          ...options,
+        };
+        const searchListeners = getSearchListeners() || [];
+        searchListeners.forEach(listener => {
+          try {
+            listener(searchValue, searchOptions, results);
+          } catch (e) {
+            console.error(e);
+          }
+        });
+        core.removeEventListener('searchInProgress', searchInProgressCallback);
+      }
+    }
+
+    function onDocumentEnd() {}
+
+    function handleSearchError(error) {
+      console.error(error);
```

Comment:
I removed the current search error handling from redux as it was not used in anywhere. Not sure what we should be doing here, though that said I never seen any errors from search. So for now I'm just logging that. Happy to change if someone has ideas what should we do with it